### ISSUE TITLE
docker-machine-driver-hyperkit behind corporate proxy: x509: certificate signed by unknown authority

### ISSUE THREAD

**Issue by gclawes** [kubernetes/minikube#7931]
**Steps to reproduce the issue:** 

1. Configure corporate proxy variables (HTTP_PROXY, HTTPS_PROXY, NO_PROXY)
2. Configure SSL_CERT_FILE env var to point to cert bundle for proxy TLS rewriting.
3. `minikube start`

**Full output of failed command:** 
```
~ $ minikube start
😄  minikube v1.9.2 on Darwin 10.15.4
✨  Automatically selected the hyperkit driver. Other choices: docker, virtualbox
💾  Downloading driver docker-machine-driver-hyperkit:
❗  Unable to update hyperkit driver: download failed: https://github.com/kubernetes/minikube/releases/download/v1.9.2/docker-machine-driver-hyperkit?checksum=file:https://github.com/kubernetes/minikube/releases/download/v1.9.2/docker-machine-driver-hyperkit.sha256: invalid checksum: Error downloading checksum file: Get "https://github.com/kubernetes/minikube/releases/download/v1.9.2/docker-machine-driver-hyperkit.sha256": x509: certificate signed by unknown authority
💿  Downloading VM boot image ...
    > minikube-v1.9.0.iso.sha256: 65 B / 65 B [--------------] 100.00% ? p/s 0s
    > minikube-v1.9.0.iso: 41.86 MiB / 174.93 MiB [ 23.93% 3.26 MiB p/s ETA 40s
```

**Full output of `minikube start` command used, if not already included:**
See above.

**Optional: Full output of `minikube logs` command:**
N/A
<details>

</details>

**Comment by tstromberg** [comment]
First, as a workaround, you can download the driver from https://github.com/kubernetes/minikube/releases and copy it into `$HOME/.minikube/bin`. You'll also need to run:

* `sudo chown root:wheel $HOME/.minikube/bin/docker-machine-driver-hyperkit`
* `sudo chmod u+s $HOME/.minikube/bin/docker-machine-driver-hyperkit`

Now as far as the underlying issue, it's difficult to tell which env variable is not being respected by Go: `SSL_CERT_FILE` or `HTTPS_PROXY`. Can you provide any additional feedback using `tcpdump` or something to isolate if the variable is being respected?

Either way, this is kind of out of minikube's hands, as it relies on Go's defaults.

**Comment by sharifelgamal** [comment]
Any update here @gclawes?

**Comment by gclawes** [comment]
Sorry for the delay.

Based on the fact that it's getting an invalid cert authority error, `SSL_CERT_FILE` is the variable it's not respecting (the proxy in question does SSL inspection, so it returns a cert signed by it's own authority).   If it were not respecting `HTTPS_PROXY` the error would be some kind of connection timeout.

```
~ [⎈ |minikube:default] $ curl -kv $HTTPS_PROXY 2>&1 | grep HTTP/
> GET / HTTP/1.1
< HTTP/1.1 400 Bad Request
~ [⎈ |minikube:default] $ curl -v $HTTPS_PROXY 2>&1 | grep HTTP/
> GET / HTTP/1.1
< HTTP/1.1 400 Bad Request
```

**Comment by gclawes** [comment]
The ISOs are released with the binaries at http://storage.googleapis.com/minikube/releases/latest?

**Comment by sharifelgamal** [comment]
The minikube ISO itself is at http://storage.googleapis.com/minikube/iso/ (e.g. http://storage.googleapis.com/minikube/iso/minikube-v1.11.0.iso), everything else is under releases/latest yes.

**Comment by gclawes** [comment]
From the original error message, it looks like it's downloading the hyperkit binary from github.com, not storage.googleapis.com: `https://github.com/kubernetes/minikube/releases/download/v1.9.2/docker-machine-driver-hyperkit`

**Comment by medyagh** [comment]
@gclawes for corp networks I strongly suggest using the docker driver, 

have you tried out newest driver Docker Driver with latest version of minikube?
you could try
minikube delete
minikube start --driver=docker

for more information on the docker driver checkout:
https://minikube.sigs.k8s.io/docs/drivers/docker/

also we had a bug on Proxy envs that we fixed in v1.11.0 you might wanna try the latest minikube
